lib/rubocop/cop/style/redundant_line_continuation.rb

```diff
@@ -87,16 +88,20 @@ def on_new_investigation
 
         def require_line_continuation?(range)
           !ends_with_backslash_without_comment?(range.source_line) ||
-            string_concatenation?(range.source_line) ||
+            string_concatenation?(range) ||
             start_with_arithmetic_operator?(processed_source[range.line])
         end
 
         def ends_with_backslash_without_comment?(source_line)
           source_line.gsub(/#.+/, '').end_with?('\\')
         end
 
-        def string_concatenation?(source_line)
-          /["']\s*\\\z/.match?(source_line)
+        def string_concatenation?(range)
+          return true if /.*["']\s*\\\z/.match?(range.source_line)
+
+          processed_source.each_token.any? do |token|
```

Comment:
`#each_token` is deprecated: https://github.com/rubocop/rubocop-ast/blob/14844626f0c6f87aef0e74198769537b40b21b7b/lib/rubocop/ast/processed_source.rb#L88-L91
```suggestion
          processed_source.tokens.each.any? do |token|
```